### ISSUE TITLE
wish: command line interface (CLI)

### ISSUE THREAD

**Issue by markstos** [GoogleChromeLabs/squoosh#276]
I would love to be able to use on on the command line.

It would be great to be able to  point it at a whole folder of images at once and see what the result is.

**Comment by trueqap** [comment]
That would be awesome

**Comment by surma** [comment]
I don’t think it makes a lot of sense to build a CLI for squoosh, considering squoosh is just running already existing CLIs via webassembly 😄 

So for now, I’d defer y’all to [OptiPNG](http://optipng.sourceforge.net), [MozJPEG](https://github.com/mozilla/mozjpeg) and [WebP](https://developers.google.com/speed/webp/). (IIRC, these are available via Brew, too).

We are, however, talking about building some form of batch-processing.

**Comment by mattmischuk** [comment]
It would be beneficial to run multiples of a given asset, I already have to do a `@1x`, `@2x`, and `@3x` version in multiple different formats. Having one CLI to do them all would be beneficial.

**Comment by Flyrell** [comment]
I agree with @mattmischuk.

Squoosh is the go-to for me to compress the images to e.g. `jpeg` and `webp`. To do that via CLI, I would need to (at least) install multiple libraries and create some sort of node script to bundle their functionality together and run the needed commands to process the image. Also, I'd want those CLI tools to output images in similar quality to what I'm seeing in Squoosh and as far as I know, it would be a bit hard to find what's going on inside the source code to make those commands run with the same settings/options as Squoosh runs them. 

It would be so useful to make it work like that:
```bash
squoosh --input image.jpg --output MozJPEG --quality 75
squoosh --input image.jpg --output WebP --quality 75 --effort 4
```

Also, it would be much easier to bundle those type of commands to run that through some directory of images, which would be one of the many possibilities it'd open.

As you said, Squoosh is already a wrapper for various CLI tools, so I don't see any disadvantages to expose the CLI API to make it work. Of course, I'd be more than happy to contribute, if you decide it's a good idea.

PS: I really appreciate the work you're doing @jakearchibald & @surma and can't wait to see another episode of HTTP 203 with your awesome duo 😄

**Comment by surma** [comment]
We may or may not be thinking about a CLI ;)

**Comment by sobujbd** [comment]
[Squoosh CLI](https://github.com/GoogleChromeLabs/squoosh/tree/dev/cli) have been proudly disclosed! Thanks for releasing **Squoosh CLI**.
